### PR TITLE
chore(auction): Moving bid extraction to from msg handler to ante handler

### PR DESCRIPTION
**Pull request by davidterpay**
### Overview
In this PR, I update the ante handler with the following changes:
1. Since the mevLane's custom `CheckTx` handler already reverifies transactions on top of the latest commit state, we do not need to worry about iterative bidding from a user. If the user wants to bid several times within a single block they can and hence the `ad.mempool.Contains(tx)` check is not necessary on recheck tx.
2. Bid extraction has been moved from the message handler to the ante handler. This is useful since bids can now be extracted upon checking a tx instead of at runtime when `runMsgs` is executed.
4. We only check a bid against the validator's mempool if the exec modes are not one of `ExecModePrepareProposal`, `ExecModeProcessProposal`, or `ExecModeFinalize`. An explanation as to why was left in the docs, but the tldr is that it can lead to poor liveness guarantees of the auction. 

### Testing
Moved most of the testing out of the ante handler into the helper funcs used from the keeper.

### CODE REVIEW

File: x/auction/ante/ante.go [hunk 1]
```diff
@@ -108,19 +108,26 @@ func (ad AuctionDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool,
...
+	if timeout == 0 && ctx.IsReCheckTx() {
+		return fmt.Errorf("timeout height cannot be 0 for recheck tx")
+	}
+
+	height := ctx.BlockHeight()
 
 	// If the mode is CheckTx or ReCheckTx, we increment the current block height by one to
 	// account for the fact that the transaction will be executed in the next block.
 	if ctx.IsCheckTx() || ctx.IsReCheckTx() {
-		currentBlockHeight++
```
[the remark this answers is not in the document]
Comment (nivasan1) [reply x/auction/ante/ante.go:116]: Like its possible that the node is CheckTx, during TimeoutCommit (before creating a proposal), in which case, the timeout would be valid?

**Review by nivasan1 — COMMENTED** [review]
lgtm outside of the ante-handler timeout check

**Review by nivasan1 — APPROVED** [review]
lgtm

File: x/auction/ante/ante.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidterpay) [reply x/auction/ante/ante.go:116]: FIxed

File: x/auction/ante/utils.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package ante
```
Comment (nivasan1) [inline x/auction/ante/utils.go:1]: nit: name this timeout.go

**Pull request merged**